src/audioPlayer.js

```diff
@@ -100,14 +109,49 @@ async function setPlayer() {
 }
 
 function resetControllers() {
-    AUDIO_TOGGLES.forEach((audioToggle) => {
-        const playIconPlayer = audioToggle.querySelector(
-            '[data-element=play]'
-        )
-        const pauseIconPlayer = audioToggle.querySelector(
-            '[data-element=pause]'
-        )
-        playIconPlayer.setAttribute('display', 'block')
-        pauseIconPlayer.setAttribute('display', 'none')
+    [...AUDIO_TOGGLES, ...VIDEO_TOGGLES].forEach((mediaToggle) => {
+        const playIcon = mediaToggle.querySelector('[data-element=play]')
+        const pauseIcon = mediaToggle.querySelector('[data-element=pause]')
+        const unmuteIcon = mediaToggle.querySelector('[data-element=unmute]')
+        const muteIcon = mediaToggle.querySelector('[data-element=mute]')
+
+        if (playIcon) playIcon.setAttribute('display', 'block')
+        if (pauseIcon) pauseIcon.setAttribute('display', 'none')
+        if (unmuteIcon) unmuteIcon.setAttribute('display', 'block')
+        if (muteIcon) muteIcon.setAttribute('display', 'none')
+    })
+}
+
+
+/**
+ * A helper function to toggle a video's mute state. Bind the trigger to this function so that the video selector can be
+ * safely retrieved and the icon in the button can be changed.
+ */
+function toggleVideoMute() {
+    const videoToggle = this
+
+    // Ensure the audio player and video audio do not play on top of each other.
+    PLAYER.pause()
+    resetControllers()
```

Comment:
For the video element that gets muted/unmuted, does the icon toggle happens two times super quickly?
The first time with `resetControllers()` and then line 143?

Reply:
There should only be two possibilities here:
- Initial state muted (icon state: muted)
- `resetControllers()` is invoked (icon state: muted)
- Lines 143/144 run (icon state: unmuted)

And:
- Initial state unmuted (icon state: unmuted)
- `resetControllers()` is invoked (icon state: muted)
- Lines 143/144 run (icon state: muted)

Either way, even though there are two separate actions changing the visibility of the icons, it won't ever go `muted -> unmuted -> muted` or `unmuted -> muted -> unmuted`.